Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp

```diff
@@ -189,7 +189,7 @@ MediaPlayerPrivateGStreamer::MediaPlayerPrivateGStreamer(MediaPlayer* player)
     m_nicosiaLayer = Nicosia::ContentLayer::create(Nicosia::ContentLayerTextureMapperImpl::createFactory(*this,
         [&]() -> Ref<TextureMapperPlatformLayerProxy> {
 #if USE(TEXTURE_MAPPER_DMABUF)
-            if (webKitDMABufVideoSinkIsEnabled() && webKitDMABufVideoSinkProbePlatform())
+            if (webKitDMABufVideoSinkProbePlatform())
```

Comment:
These two conditions should match conditions in `createVideoSinkDMABuf()`. Otherwise things can get weird with using TextureMapperPlatformLayerProxyDMABuf along with a GL sink.

So I would recommend keeping `webKitDMABufVideoSinkIsEnabled()`, and using `WEBKIT_GST_DISABLE_DMABUF_SINK` env to disable it. That way behavior between DMABuf sink and GL sink can be examined by selectively disabling one or the other, as long as they coexist.

Reply:
Makes sense! I'll update the PR, thanks for the review.

Reply:
Maybe we can make this work with `GST_PLUGIN_FEATURE_RANK=webkitdmabufsink:0` too. I can try that, I've been wondering lately about avoid new env vars when we can...

Reply:
GST_PLUGIN_FEATURE_RANK works only during `gst_init_check()` so for our custom elements registered afterwards we'd need to re-parse the env var... Not sure it's worth.